cmake/configure/configure_symengine.cmake

```diff
@@ -0,0 +1,43 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2019 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+#
+# Configuration for the SymEngine library:
+#
+
+#
+# We require at least version 0.3 of the symengine library:
+#
+SET(SYMENGINE_MINIMUM_REQUIRED_VERSION "0.3")
+
+
+MACRO(FEATURE_SYMENGINE_CONFIGURE_EXTERNAL)
+  SET(DEAL_II_SYMENGINE_WITH_LLVM ${SYMENGINE_WITH_LLVM})
+
+  IF(DEAL_II_SYMENGINE_WITH_LLVM)
+    MESSAGE(STATUS "Configured with SymEngine LLVM capabilities.")
+  ENDIF()
+
+  #
+  # Remove a flag from the list compiled by SymEngine.
+  # This prevents us from compiling in C++11 mode even if we've previously
+  # determined that we will be compiling with C++14 support enabled.
+  #
+  REPLACE_FLAG(SYMENGINE_CXX_FLAGS_DEBUG "-std=c++11" "")
+  REPLACE_FLAG(SYMENGINE_CXX_FLAGS_RELEASE "-std=c++11" "")
```

Comment:
Do we really need to use `SYMENGINE_CXX_FLAGS_DEBUG/RELEASE`?
On my system this sets for example
```
SYMENGINE_CXX_FLAGS_RELEASE = -Wno-unused-parameter -march=native -fopenmp
SYMENGINE_CXX_FLAGS_DEBUG = -Wall -Wextra -fno-common -fPIC -g -Wno-unused-parameter -ggdb -fopenmp
```
This doesn't look terribly useful to me.
What about simply zeroing both vectors?
```
SET(SYMENGINE_CXX_FLAGS_DEBUG)
SET(SYMENGINE_CXX_FLAGS_RELEASE)
```

Reply:
We might. They also link to a whole bunch of libraries. Surely we'd need to add those link flags too if `RPATH` is not used?

Reply:
Oh, I've been silly. Those aren't the link flags... Yes, lets just zero them.